dataprofiler/profilers/graph_profiler.py

```diff
@@ -220,60 +225,88 @@ def update(self, graph: nx.Graph) -> GraphProfiler:
     """
 
     def _update_num_nodes(
-        self, graph, prev_dependent_properties=None, subset_properties=None
-    ):
+        self,
+        graph: nx.Graph,
+        prev_dependent_properties: Dict = None,
+        subset_properties: Dict = None,
+    ) -> None:
         """Update num_nodes for profile."""
         self._num_nodes = self._get_num_nodes(graph)
-        subset_properties["num_nodes"] = self._num_nodes
+        subset_properties["num_nodes"] = self._num_nodes  # type: ignore
 
     def _update_num_edges(
-        self, graph, prev_dependent_properties=None, subset_properties=None
-    ):
+        self,
+        graph: nx.Graph,
+        prev_dependent_properties: Dict = None,
+        subset_properties: Dict = None,
+    ) -> None:
         """Update num_edges for profile."""
         self._num_edges = self._get_num_edges(graph)
 
     def _update_avg_node_degree(
-        self, graph, prev_dependent_properties=None, subset_properties=None
-    ):
+        self,
+        graph: nx.Graph,
+        prev_dependent_properties: Dict = None,
+        subset_properties: Dict = None,
+    ) -> None:
         """Update avg_node_degree for profile."""
         self._avg_node_degree = self._get_avg_node_degree(
-            graph, subset_properties["num_nodes"]
+            graph, subset_properties["num_nodes"]  # type: ignore
```

Comment:
why do we need `# type: ignore` here?

Reply:
`subset_properties`'s default value is `None` which is not subscriptable

Reply:
Is there ever a chance where that could get passed to here?

Reply:
I think `{}` is usually passed in for `subset_properties`, so I changed the default value from `None` to `{}` and removed all of the `# type: ignore`.